### DISCUSSION TITLE
fsh-api-animal features

### DISCUSSION THREAD

**Discussion by Malix-Labs** [scratch-for-discord/Web-Application_Frontend#203] [Q&A] [answered]
What does the [fsh-api-animal](https://github.com/scratch-for-discord/Web-Application_Frontend/blob/af2c1227224bd6c4cc93d553349cde99be85bad4/src/blocks/invention/fsh-api-animal.js) block(s) features?

**Comment by inventionpro** [comment, marked as answer]
with inv_fsh_api_animal you input an animal and outputs a random image of that animal
you can see list of animal on it's help url
also that block is made by me and frozer #200

**Reply by Malix-Labs** [reply]
A "random" image, or an AI generated image?

**Reply by inventionpro** [reply]
"random", no ai

**Reply by inventionpro** [reply]
also mind accepting pull request #201

**Reply by JustLarsHere** [reply]
isnt that you basically being lazy and making a block for your own bot?

**Reply by Malix-Labs** [reply]
> isnt that you basically being lazy and making a block for your own bot?

?
What do you mean and what are you talking about?

**Reply by JustLarsHere** [reply]
i mean that the block, basically is http calling his bot for an image. im unsure why anyone would need that if they could make it themselves

**Reply by JeremyGamer13** [reply]
isnt it frost's repl

**Reply by JustLarsHere** [reply]
it comes from the fsh bot, (a bot made by frozer, and inv and partially me) i just dont see the reason, to have a block that, gets an image from that bot.

**Reply by JustLarsHere** [reply]
cause, everytime the bot is down, that block also wont work

**Reply by JustLarsHere** [reply]
its like an extra step to the api that fsh is using

**Reply by Malix-Labs** [reply]
> i mean that the block, basically is http calling his bot for an image. im unsure why anyone would need that if they could make it themselves

It's called a function, not a feature.

S4D as a whole is a function.

**Comment by nekoify** [comment]
if only there was a certain block that would let you make api requests

**Reply by Malix-Labs** [reply]
[about that...](https://discord.com/channels/932651844344373278/1084251074736693349/1084251074736693349)
